stan/math/prim/scal/fun/lgamma.hpp

```diff
@@ -89,6 +89,17 @@ inline double lgamma(int x) {
 #endif
 }
 
+inline double lgamma(unsigned int x) {
+#if !__MINGW32__
+  int sign = 1;
+  return ::lgamma_r(x, &sign);
```

Comment:
I have literally no idea :-) I just copied the code for `inline double lgamma(int x)`, (just above in the file) all the quirks are inherited from there. The code seems to have been written by @wds15 in https://github.com/stan-dev/math/pull/1255. It is possible that duplicating the code was not the best way to do this. Alternatively, I can probably force conversion to signed `int` somewhere and remove this call, but that felt less elegant.

Reply:
> The logic is simpler if the loops are rearranged and the negation is eliminated.

Not sure what you are referring to - perhaps you ment a diffent chunk of code?

Reply:
If there's cut-and-pasted code, please cross-link so we can fix both at the same time.  Hopefully @wds15 can explain what's going on and we can either leave the cut and paste or change them both later.

Reply:
I meant the logic's easier on the reader without the negation.  In general, it's easier to follow

```
#if C
...C is true...
#else
...C is false...
```

than

```
#if !C
...C is false...
#else
... C is true...
```

On the other hand, we usually want to peel off exceptional behavior so that the main line of execution is not nested:

```
if (bad input) {
  ... handle with return or excpetion
}
... main code ...
```

Reply:
This is the C library version of lgamma that is reentrant safe as opposed to `std::lgamma`. 

See https://github.com/stan-dev/math/pull/1255 for details, but the main point is: 

> The lgamma_r is a C library call which is only available upon defining -D_REENTRANT, but this is not working on Windows with MinGW32. This is why the boost version is used on Windows.

Reply:
Turns out, I only needed the copy-pasted code because I was using `unsigned int` in my tests, but all of Stan uses signed ints, so I was able to remove the code.